trin-core/src/portalnet/overlay.rs

```diff
@@ -192,12 +202,156 @@ where
     }
 
     /// Process overlay uTP listener event
-    pub fn process_utp_event(&self, event: UtpListenerEvent) {
-        // TODO: Handle overlay uTP events
-        debug!(
-            "Got overlay uTP event: {event:?}, protocol id: {:?}",
-            self.protocol
-        );
+    pub fn process_utp_event(&self, event: UtpListenerEvent) -> anyhow::Result<()> {
+        match event {
+            UtpListenerEvent::ClosedStream(utp_payload, _, stream_id) => {
+                // Handle all closed uTP streams, currently we process only AcceptStream here.
+                // FindContent payload is processed explicitly when we send FindContent request.
+                // We don't expect to process a payload for ContentStream and OfferStream.
+                match stream_id {
+                    UtpStreamId::AcceptStream(content_keys) => {
+                        self.process_accept_utp_payload(content_keys, utp_payload)?
+                    }
+                    UtpStreamId::ContentStream(_) => {}
+                    UtpStreamId::FindContentStream => {}
+                    UtpStreamId::OfferStream => {}
```

Comment:
Maybe worth a short `warn!()` on `ContentStream` and `OfferStream` then, since it's an unexpected code path.

Reply:
No need to warn here because we will receive `OfferStream` and `ContentStream` events but we don't want to process the payload because there should be no payload received (those streams are used to send uTP DATA packets). Current uTP listener events are maybe not with the optimal structure but we can refactor this at some point.